File: LC/Python/Medium/3016.py
class Solution:
    def minimumPushes(self, word: str) -> int:
        count = Counter(word)
        
        heap = [-freq for freq in count.values()]
        heapq.heapify(heap)
        
        ans = 0
        i = 0
        
        while heap:
            ans += (1 + i // 8) * -heappop(heap)
            i += 1
            
        return ans

File: LC/C++/Medium/3016.cpp
class Solution {
public:
    int minimumPushes(string word) {
        std::vector<int> frequency(26);

        for (char c : word) {
            frequency[c - 'a']++;
        }

        std::sort(frequency.begin(), frequency.end(), std::greater<int>{});

        int ans = 0;

        for (int i = 0; i < 26 && frequency[i] > 0; i++) {
            ans += frequency[i] * (i / 8 + 1);
        }

        return ans;
    }
};